### PR TITLE
Fix parsing of charm URLs in the charm model.

### PR DESCRIPTION
**Pull request by frankban**
This way, we reuse the existing URL library and we allow users with the "+" sign, as they are accepted by juju and USSO.

Also add a missing component property declaration.

### CODE REVIEW

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7723/

**Comment by frankban** [comment]
QA: as this is a change to the charm model layer, please navigate to the search view, to charms and bundles, try to deploy them, check that charm info is correctly displayed in detail pages and the inspector.

File: jujugui/static/gui/src/app/components/inspector/inspector.js [hunk 1]
```diff
@@ -144,7 +144,8 @@ YUI.add('inspector-component', function() {
               service={service}
               serviceRelations={this.props.serviceRelations}
               showActivePlan={this.props.showActivePlan}
-              showPlans={this.props.showPlans} />,
+              showPlans={this.props.showPlans}
+            />,
```
Comment (Lukewh) [inline jujugui/static/gui/src/app/components/inspector/inspector.js]: I agree, closing tags should be on a newline.

**Comment by Lukewh** [comment]
## QA OK

Though I've noticed an issue with the sticky header for bundles: https://github.com/juju/juju-gui/issues/2944

File: jujugui/static/gui/src/app/components/inspector/service-overview/service-overview.js [hunk 2]
```diff
@@ -33,7 +33,8 @@ YUI.add('service-overview', function() {
       modelUUID: React.PropTypes.string.isRequired,
       service: React.PropTypes.object.isRequired,
       serviceRelations: React.PropTypes.array.isRequired,
-      showActivePlan: React.PropTypes.func.isRequired
+      showActivePlan: React.PropTypes.func.isRequired,
+      showPlans: React.PropTypes.bool.isRequired
```
Comment (urosj) [inline jujugui/static/gui/src/app/components/inspector/service-overview/service-overview.js]: This doesn't seem related to how URLs are parsed. The name of the PR is misleading.

**Comment by frankban** [comment]
Thanks for the reviews!
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge
